sites/geohub/src/lib/RasterTileData.ts

```diff
@@ -3,21 +3,32 @@ import {
 	getActiveBandIndex,
 	getDefaltLayerStyle,
 	getDefaltLayerStyleForStac,
+	getDefaultLayerStyleForStacProducts,
 	getFirstSymbolLayerId
 } from './helper';
 import type { RasterTileMetadata, DatasetFeature, LayerCreationInfo } from './types';
 import type { Map } from 'maplibre-gl';
 
 export class RasterTileData {
 	private feature: DatasetFeature;
+	private product?: string;
 
-	constructor(feature: DatasetFeature) {
+	constructor(feature: DatasetFeature, product?: string) {
 		this.feature = feature;
+		this.product = product;
 	}
 
 	public getMetadata = async () => {
 		const metadataUrl = this.feature.properties?.links?.find((l) => l.rel === 'info').href;
 		if (!metadataUrl) return;
+		if (this.product) {
+			const res = await fetch(metadataUrl);
+			const metadata_json = await res.json();
+			const assets = Object.keys(metadata_json);
+			const firstAsset = assets[0];
+			const metadata: RasterTileMetadata = metadata_json[firstAsset];
+			return metadata;
+		}
 		const res = await fetch(metadataUrl);
 		const metadata: RasterTileMetadata = await res.json();
 		if (metadata && metadata.band_metadata && metadata.band_metadata.length > 0) {
```

Comment:
Line no 24 and 25 can be moved before if statement. it is duplicated with line no. 32 and 33

```js
const res = await fetch(metadataUrl);
const metadata_json = await res.json();
```

Reply:
I have moved this line 
`const res = await fetch(metadataUrl);`

`const metadata_json = await res.json();` can't be moved outside `if` statement as it handles a different object